cmd/leptond/main.go

```diff
@@ -118,29 +108,26 @@ func runMain() error {
 			camera.Close()
 		}
 	}()
+
 	for {
-		camera, err = lepton3.New(conf.SPISpeed)
+		camera, err = initialiseLepton(conf.SPISpeed)
 		if err != nil {
 			return err
 		}
-		camera.SetLogFunc(func(t string) { log.Printf(t) })
-
-		log.Print("enabling radiometry")
-		if err := camera.SetRadiometry(true); err != nil {
-			return err
-		}
-
-		log.Print("opening camera")
-		if err := camera.Open(); err != nil {
-			return err
-		}
 
-		err := runCamera(conf, camera, conn)
-		if err != nil {
-			if _, isNextFrameErr := err.(*nextFrameErr); !isNextFrameErr {
-				return err
+		for {
+			err = runCamera(conf, camera)
+			if err != nil {
+				if isConnectionError(err) {
+					log.Print(err.Error())
```

Comment:
No need for the call to Error. `Print` and friends will notice that the thing they're printing conforms to the error interface and will do this for you.

Reply:
oh yea :P